### PR TITLE
Added functionality to push daily reminder notification everyday at 1…

### PR DESCRIPTION
**Pull request by hariom982**
…2AM urging to fill in expenses ,transactions and incomes;

## Pull request (PR) checklist

Please check if your pull request fulfills the following requirements:

- [ ] The PR is submitted to the `main` branch.
- [ ] I've read the [Contribution Guidelines](https://github.com/Ivy-Apps/ivy-wallet/blob/main/CONTRIBUTING.md) and my PR doesn't break the rules.
- [ ] I've read and understand the [Developer Guidelines](https://github.com/Ivy-Apps/ivy-wallet/blob/main/docs/Guidelines.md).
- [ ] I confirm that I've run the code locally and everything works as expected.
- [ ] My PR includes only the necessary changes to fix the issue (i.e., no unnecessary files or lines of code are changed).
- [ ] 🎬 I've attached a **screen recording** of using the new code to the next paragraph (if applicable).

## What's changed?
Describe with a few bullets **what's new:**

- ...

- ...

## Risk factors

**What may go wrong if we merge your PR?**

- ...
- ...

**In what cases won't your code work?**

- ...
- ...

## Does this PR close any GitHub issues?

- Closes #{ISSUE_NUMBER}

## Troubleshooting CI failures ❌

Pull request checks failing? Read our [CI Troubleshooting guide](https://github.com/Ivy-Apps/ivy-wallet/blob/main/docs/CI-Troubleshooting.md).

### CODE REVIEW

File: .idea/inspectionProfiles/Project_Default.xml
Comment (ILIYANGERMANOV) [inline .idea/inspectionProfiles/Project_Default.xml:1]: Revert this change

File: build.gradle.kts
Comment (ILIYANGERMANOV) [inline build.gradle.kts:1]: Revert these changes

File: app/src/main/java/com/ivy/wallet/RootActivity.kt [hunk 1]
```diff
@@ -465,4 +469,78 @@ class RootActivity : AppCompatActivity(), RootScreen {
         val addTransactionWidget = ComponentName(this, widget)
         appWidgetManager.requestPinAppWidget(addTransactionWidget, null, null)
     }
+
+}
+//For pushing Notification to the user as a reminder at every 12AM
+// urging to fill in expenses, transactions and incomes
+fun createNotificationChannel(context: Context) {
```
Comment (ILIYANGERMANOV) [inline app/src/main/java/com/ivy/wallet/RootActivity.kt:476]: Extract this logic in a separate class and module that lives outside `:app`. For example, `:feature:reminders`

File: app/src/main/java/com/ivy/wallet/RootActivity.kt [hunk 2]
```diff
@@ -99,6 +99,10 @@ class RootActivity : AppCompatActivity(), RootScreen {
         setupDatePicker()
         setupTimePicker()
 
+        createNotificationChannel(this)
```
Comment (ILIYANGERMANOV) [inline app/src/main/java/com/ivy/wallet/RootActivity.kt:102]: Not all users might want to have reminders, use `IvyFeatures` to make it a customizable setting that's enabled by default

**Review by ILIYANGERMANOV — CHANGES_REQUESTED** [review]
1. Fill the PR template 
2. Revert unnecessary changes
3. Extract the reminders feature as a separate module
4. Make the reminder a Setting that users can turn off

File: app/src/main/java/com/ivy/wallet/RootActivity.kt [hunk 3]
```diff
@@ -465,4 +469,78 @@ class RootActivity : AppCompatActivity(), RootScreen {
...
+        .setContentText("Fill in todays expences,transactions and incomes.")
+        .setPriority(NotificationCompat.PRIORITY_HIGH)
+        .setContentIntent(pendingIntent)
+        .setAutoCancel(true)
+
+    with(NotificationManagerCompat.from(context)) {
+        if (ActivityCompat.checkSelfPermission(
+                context,
+                Manifest.permission.POST_NOTIFICATIONS
+            ) != PackageManager.PERMISSION_GRANTED
```
Comment (ILIYANGERMANOV) [inline app/src/main/java/com/ivy/wallet/RootActivity.kt:508]: The code would simpler if:
1. Extract the permission check as a `hasNotificationsPermission` function
2. if (hasNotificationsPermisson()) notify

File: app/src/main/java/com/ivy/wallet/RootActivity.kt [hunk 4]
```diff
@@ -465,4 +469,78 @@ class RootActivity : AppCompatActivity(), RootScreen {
...
+fun sendNotification(context: Context) {
+    val intent = Intent(context, MainActivity::class.java).apply {
+        flags = Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TASK
+    }
+    val pendingIntent: PendingIntent = PendingIntent.getActivity(context, 0, intent,
+        PendingIntent.FLAG_IMMUTABLE)
+
+    val builder = NotificationCompat.Builder(context, "DAILY_REMINDER_CHANNEL_ID")
+        .setSmallIcon(R.drawable.ic_launcher-playstore)
+        .setContentTitle("EveryDay Reminder")
```
Comment (ILIYANGERMANOV) [inline app/src/main/java/com/ivy/wallet/RootActivity.kt:498]: Extract these a string so they can be localized

**Comment by ILIYANGERMANOV** [comment]
Overall, the implementation looks good and should work but please attach a screen recording testing it

File: app/src/main/java/com/ivy/wallet/RootActivity.kt [hunk 5]
```diff
@@ -465,4 +469,78 @@ class RootActivity : AppCompatActivity(), RootScreen {
...
+        .setContentText("Fill in todays expences,transactions and incomes.")
```
Comment (suyash01) [inline app/src/main/java/com/ivy/wallet/RootActivity.kt:499]: typo in `expenses`

File: app/src/main/java/com/ivy/wallet/RootActivity.kt [hunk 6]
```diff
@@ -465,4 +469,78 @@ class RootActivity : AppCompatActivity(), RootScreen {
...
+fun scheduleDailyReminder(context: Context) {
+    val alarmManager = context.getSystemService(Context.ALARM_SERVICE) as AlarmManager
+    val intent = Intent(context, ReminderBroadcastReceiver::class.java)
+    val pendingIntent = PendingIntent.getBroadcast(context, 0, intent,
+        PendingIntent.FLAG_IMMUTABLE)
+
+    val calendar: Calendar = Calendar.getInstance().apply {
+        timeInMillis = System.currentTimeMillis()
+        set(Calendar.HOUR_OF_DAY, 12)
+        set(Calendar.MINUTE, 00)
```
Comment (suyash01) [inline app/src/main/java/com/ivy/wallet/RootActivity.kt:524]: Having configurable time of the day via settings will be more desirable.

Comment (ILIYANGERMANOV) [reply app/src/main/java/com/ivy/wallet/RootActivity.kt:524]: +1

Comment (hariom982) [reply app/src/main/java/com/ivy/wallet/RootActivity.kt:524]: would work on it but might take time
